server/src/main/java/org/elasticsearch/search/aggregations/metrics/CollectedMax.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.aggregations.metrics;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.DoubleArray;
+import org.elasticsearch.search.aggregations.AggregationReduceContext;
+import org.elasticsearch.search.aggregations.CollectedAggregator;
+import org.elasticsearch.search.aggregations.InternalAggregation;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class CollectedMax extends CollectedAggregator {
+    private DoubleArray maxes;
+
+    protected CollectedMax(String name, Map<String, Object> metadata, BigArrays bigArrays, long size) {
+        super(name, metadata, bigArrays);
+        maxes = bigArrays().newDoubleArray(size, false);
+    }
+
+    public CollectedMax(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public String getWriteableName() {
+        return MaxAggregationBuilder.NAME;
+    }
+
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return null;
```

Comment:
Probably worth a NOCOMMIT. It has to be whatever version we land. I tend to compile for a while with `./gradlew -p server check -x forbiddenPatterns` or something like that. That way `NOCOMMIT` can block merging but doesn't block `precommit`.